core/primitives/src/epoch_sync.rs

```diff
@@ -70,16 +69,25 @@ pub struct EpochSyncProofEpochData {
     /// The block producers and their stake, for this epoch. This is verified
     /// against the `next_bp_hash` of the `last_final_block_header` of the epoch before this.
     pub block_producers: Vec<ValidatorStake>,
+    /// Whether the block producers are encoded in the old format for computing the bp_hash.
+    /// The encodings between old and new format do not collide, so this field does not need
```

Comment:
May be worth mentioning why iterator over validator stake can't be encoded as iterator over validator stake v1 and vice versa. Looks like because account id can't start with zero byte.

Reply:
Turns out it's not that trivial to argue - I've included a comment.